src/transformers/models/bloom/modeling_bloom.py

```diff
@@ -230,106 +227,130 @@ def __init__(self, config, layer_number=None):
             )
 
         # Layer-wise attention scaling
-        self.layer_number = max(1, layer_number)
-        self.norm_factor = math.sqrt(self.head_dim) * self.layer_number
+        self.inv_norm_factor = 1.0 / math.sqrt(self.head_dim)
+        self.beta = 1.0
```

Comment:
This change is because we did not use `apply_query_key_layer_scaling` for the training of BLOOM models?

Reply:
No we did, but I actually don't think we need it. I think the trick was used to be able to train fp16 models, but since our fp16 models are small and the big ones are bf16 that trick shouldn't be use anymore IMO.

Reply:
But don't we still need it then, as the checkpoints are trained with layer scaling?

Reply:
Why? So that you mimick perfectly the logits? Not sure this matters, and we should make sure to obtain the maximum precision we can get from the weights IMO. I'm running a bunch of greedy_search on 175b right now to assess if that matters.